vars/packer.groovy

```diff
@@ -122,6 +124,16 @@ def call(body) {
     throw new GroovyRuntimeException("Unable to find AMI")
   }
 
+  if (config.ebsDeviceName) {
+    ebsDeviceName = config.ebsDeviceName
+  }
+  if (config.ebsVolumeSize) {
+    ebsVolumeSize = config.ebsVolumeSize
+  }
+  if ((ebsDeviceName && !ebsVolumeSize) || (!ebsDeviceName && ebsVolumeSize)) {
+    throw new GroovyRuntimeException("Supply both ebs Volumes Size and DeviceName")
+  }
+
```

Comment:
you could simplify this logic to as `config.ebsDeviceName` will return null if not set

```groovy
if ((config.ebsDeviceName && !config.ebsVolumeSize) || (!config.ebsDeviceName && config.ebsVolumeSize)) {
    throw new GroovyRuntimeException("Supply both ebs Volumes Size and DeviceName to set a volume in the packer template")
}

if (config.ebsDeviceName && config.ebsVolumeSize) {
    ptb.addBlockDevice(config.ebsDeviceName, config.ebsVolumeSize)
}

Reply:
Change applied